src/three/PNTSLoader.js

```diff
@@ -1,53 +1,84 @@
 import { PNTSLoaderBase } from '../base/PNTSLoaderBase.js';
-import { Points, PointsMaterial, BufferGeometry, BufferAttribute, DefaultLoadingManager } from 'three';
-
+import {
+	Points,
+	PointsMaterial,
+	BufferGeometry,
+	BufferAttribute,
+	DefaultLoadingManager,
+} from 'three';
+
+const DRACO_ATTRIBUTE_MAP = {
+	RGB: 'color',
+	POSITION: 'position',
+};
 export class PNTSLoader extends PNTSLoaderBase {
 
 	constructor( manager = DefaultLoadingManager ) {
 
 		super();
 		this.manager = manager;
 
+		// hacky way of getting the draco loader from the manager
+		this.dracoLoader = this.manager.getHandler( 'draco.drc' );
+
 	}
 
 	parse( buffer ) {
 
-		return super
-			.parse( buffer )
-			.then( result => {
+		return super.parse( buffer ).then( async ( result ) => {
 
-				const { featureTable } = result;
+			const { featureTable } = result;
 
-				const POINTS_LENGTH = featureTable.getData( 'POINTS_LENGTH' );
-				const POSITION = featureTable.getData( 'POSITION', POINTS_LENGTH, 'FLOAT', 'VEC3' );
-				const RGB = featureTable.getData( 'RGB', POINTS_LENGTH, 'UNSIGNED_BYTE', 'VEC3' );
+			const geometry = new BufferGeometry();
+			const material = new PointsMaterial();
 
-				[
-					'QUANTIZED_VOLUME_OFFSET',
-					'QUANTIZED_VOLUME_SCALE',
-					'CONSTANT_RGBA',
-					'BATCH_LENGTH',
-					'POSITION_QUANTIZED',
-					'RGBA',
-					'RGB565',
-					'NORMAL',
-					'NORMAL_OCT16P',
-				].forEach( feature => {
+			if ( featureTable.isDraco() ) {
 
-					if ( feature in featureTable.header ) {
+				const dracoIDs = featureTable.getDracoProperties();
+				const attributeIDs = {};
 
-						console.warn( `PNTSLoader: Unsupported FeatureTable feature "${ feature }" detected.` );
+				for ( const [ key, value ] of Object.entries( dracoIDs ) ) {
 
-					}
+					attributeIDs[ DRACO_ATTRIBUTE_MAP[ key ] ] = value;
 
-				} );
+				}
```

Comment:
This will cause an undefined key if a DRACO property exists that is not "RGB" or "POSITION", right?